Products/CMFPlone/MigrationTool.py

```diff
@@ -88,6 +88,7 @@ def upgrade_all(self, context):
 ADDON_LIST = AddonList([
     Addon(profile_id='Products.CMFEditions:CMFEditions'),
     Addon(profile_id='Products.CMFPlacefulWorkflow:CMFPlacefulWorkflow'),
+    Addon(profile_id='plone.app.caching:default'),
```

Comment:
I think you need to initialize `Addon` it with `check_module=True`, because it is not guaranteed that `plone.app.caching` is there. Like if one does not need it and only depends on `Products.CMFPlone`, instead of `Plone`.

```suggestion
    Addon(profile_id='plone.app.caching:default', check_module=True),
```

Reply:
Good point.
Ah, but the same is true for `CMFPlacefulWorkflow` and `plone.app.iterate`, which are also in this list.
In fact, we do not need this check for these packages. I tried it out with a wrongly spelled name, and the only result is a warning from GenericSetup:

    Version of profile plone.app.katching:default is unknown, refusing to upgrade.

This is the same warning that you get for the others, unless you have actually installed them.

Reply:
Actually, if we use `check_module`, and the module cannot be imported, we get an info message, which is better than a warning. So let me change this anyway.
I will do the same for a few other packages.

Reply:
Done.
And I replaced `plone.app.iterate:plone.app.iterate` with `plone.app.iterate:default`.